oak-segment-remote/src/main/java/org/apache/jackrabbit/oak/segment/remote/persistentcache/PersistentDiskCache.java

```diff
@@ -204,7 +204,17 @@ private void cleanUpInternal() {
                     }
                     if (cacheSize.get() > maxCacheSizeBytes * 0.66) {
                         File segment = segmentCacheEntry.getPath().toFile();
-                        cacheSize.addAndGet(-segment.length());
+                        long length = segment.length();
+                        if (length == 0) {
+                            if (logger.isDebugEnabled()) {
+                                logger.debug("Avoiding removal of zero-sized file {}", segmentCacheEntry.getPath());
+                            } else {
+                                logger.warn("Avoiding removal of zero-sized file.");
+                            }
+                            return;
+                        }
+                        long cacheSizeAfter = cacheSize.addAndGet(-length);
+                        diskCacheIOMonitor.updateCacheSize(cacheSizeAfter, directory.length());
```

Comment:
https://docs.oracle.com/javase/8/docs/api/java/io/File.html#length--

API docs says that File.length returns unspecified value for directories.

Reply:
It probably should be this:

```suggestion
                        diskCacheIOMonitor.updateCacheSize(cacheSizeAfter, segment.length());
```

Reply:
You're both right, the directory size can only be calculated by traversing the directory, the size attribute only shows the size of the directory structure itself on disk. I'm going to change this to measure the calculated size and the size change instead.